### PR TITLE
Add ability to disable logging to stdout

### PR DESCRIPTION
**Pull request by farmerbb**
Guard all calls to `println()` behind a flag `enableLogging` that can be set on the GBSDKBuilder.  This cleans up unecessary log output on production apps using the SDK.

### CODE REVIEW

**Comment by Bohdan-Kim** [comment]
The changes look good

File: Core/build.gradle.kts [hunk 1]
```diff
@@ -48,6 +48,10 @@ android {
             isMinifyEnabled = false
         }
     }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_11
+        targetCompatibility = JavaVersion.VERSION_11
```
[the remark this answers is not in the document]
Comment (farmerbb) [reply Core/build.gradle.kts]: This has now been removed

File: Core/src/commonMain/kotlin/com/sdk/growthbook/network/NetworkDispatcher.kt [hunk 2]
```diff
@@ -25,4 +25,6 @@ interface NetworkDispatcher {
         onSuccess: (String) -> Unit,
         onError: (Throwable) -> Unit
     )
+
+    fun setLoggingEnabled(enabled: Boolean)
```
Comment (Bohdan-Kim) [inline Core/src/commonMain/kotlin/com/sdk/growthbook/network/NetworkDispatcher.kt:29]: We already have compiled binaries with the implementation of NetworkDispatcher. For me using current version of OkHttp dispatcher causes runtime error `java.lang.AbstractMethodError`. I'll rewrite this code without changing NetworkDispatcher interface but from my curiosity can you advise something in this case. Seems like default implementation doesn't help some why

Comment (Bohdan-Kim) [reply Core/src/commonMain/kotlin/com/sdk/growthbook/network/NetworkDispatcher.kt:29]: @JvmDefault annotation is not available in commonMain source set

**Pull request merged**